api/src/AppBundle/Controller/UserController.php

```diff
@@ -251,17 +254,20 @@ public function getUserTeamNames(Request $request, $email)
      * //TODO move to UserService
      *
      * @Route("/{id}", methods={"DELETE"})
-     * @Security("has_role('ROLE_ADMIN')")
+     * @Security("has_role('ROLE_ADMIN') or has_role('ROLE_SUPER_ADMIN')")
      *
      * @param int $id
+     * @return array
+     * @throws ORMException
+     * @throws OptimisticLockException
      */
     public function delete($id)
     {
         /** @var User $user */
         $user = $this->userRepository->find($id);
 
-        if ($user->getRoleName() !== User::ROLE_LAY_DEPUTY) {
-            throw $this->createAccessDeniedException('Cannot delete users with role ' . $user->getRoleName());
+        if (!in_array($user->getRoleName(),self::DELETABLE_ROLES)) {
+            throw $this->createAccessDeniedException('Cannot delete ' . $user->getRoleName() . ' users');
```

Comment:
I'm a little on the fence with this, but from what I can tell, the restriction as to which roles can delete which users is encapsulated in the API consumer (client), instead of/along with the API itself. If we were to unit test this endpoint in isolation, or hit it via postman, the requirement to restrict deletion would not be met. I don't know if that's an issue, so any thoughts would be good.

Reply:
Yeah that's a good point. We protect against unauthorised roles accessing the endpoint from the client but theres no protection for future changes to the code at the API level. It feels a bit off duplicating the logic in API and client but they are useful to have on both sides...what do you think?

Reply:
I would opt to duplicate the logic. It's a PITA but means the rules are at the lowest level. I can imagine a future where we have an additional client app and it starts deleting users it shouldn't.

Reply:
I tend to agree. Where practical, I prefer to write methods that protect themselves by making no assumptions about the context in which it was called, so if we can have the buck stop here that would be good.